File: 18_정지윤/10828(2).java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    int[] stack = new int[10000];
    int num = -1;

    void push(int X) {
        num++;
        stack[num] = X;
    }
    void pop() {
        if (num == -1) {
            System.out.println(-1);
        } else {
            int X = stack[num];
            num--;
            System.out.println(X);
        }
    }
    void size() {
        System.out.println(num + 1);
    }
    void empty() {
        if (num == -1) {
            System.out.println(1);
        } else{
            System.out.println(0);
        }
    }
    void top() {
        if (num == -1) {
            System.out.println(-1);
        } else {
            System.out.println(stack[num]);
        }
    }
    private void solve() {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            int N = Integer.parseInt(br.readLine());
            String[] s;
            for (int i = 0; i < N; ++i) {
                s = br.readLine().split(" ");

                if (s[0].equals("push")) {
                    int X = Integer.parseInt(s[1]);
                    push(X);
                } else if (s[0].equals("pop")) {
                    pop();
                } else if (s[0].equals("size")) {
                    size();
                } else if (s[0].equals("empty")) {
                    empty();
                } else if (s[0].equals("top")){
                    top();
                }
            }
        }catch (IOException e) {
            e.printStackTrace();
        }
    }
    public static void main(String[] args){
        new Main().solve();
    }
}

File: 15_이준용/10828(2).py
from sys import stdin
input = stdin.readline
class Stack:
    stack = []
    def push(self, num):
        self.stack.append(num)
    def pop(self):
        print(self.stack.pop()) if self.stack else print(-1)
    def size(self):
        print(len(self.stack))
    def empty(self):
        print(0) if self.stack else print(1)
    def top(self):
        print(self.stack[-1]) if self.stack else print(-1)
    func = {
        'push' : push,
        'pop' : pop,
        'size' : size,
        'empty' : empty,
        'top' : top
    }

stk = Stack
for _ in range(int(input())):
    comm= input().split()
    if len(comm)==2:
        stk.func[comm[0]](stk, int(comm[1]))
    else:
        stk.func[comm[0]](stk)